### PR TITLE
Update Azure OpenAI to 2024-04-01-preview.

### PR DESCRIPTION
**Pull request by glecaros**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to module CHANGELOG.md are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

File: sdk/ai/azopenai/CHANGELOG.md [hunk 1]
```diff
@@ -1,17 +1,15 @@
 # Release History
 
-## 0.5.2 (Unreleased)
+## 0.5.2 (2024-05-07)
```
Comment (jhendrixMSFT) [inline sdk/ai/azopenai/CHANGELOG.md:3]: Looks like there were some breaking changes, so this should bump to `0.6.0`.

Comment (richardpark-msft) [reply sdk/ai/azopenai/CHANGELOG.md:3]: I don't think there are any breaking changes (all my old tests passed without modification). 

If you're talking about the customError, that has to do with the test resources being oversubscribed, but it's not a contract change.

**Review by richardpark-msft — APPROVED** [review]

File: sdk/ai/azopenai/models.go [hunk 2]
```diff
@@ -1328,7 +1358,7 @@ type ContentFilterResult struct {
 // ContentFilterResultDetailsForPrompt - Information about content filtering evaluated against input data to Azure OpenAI.
 type ContentFilterResultDetailsForPrompt struct {
 	// Describes detection results against configured custom blocklists.
-	CustomBlocklists []ContentFilterBlocklistIDResult
+	CustomBlocklists *ContentFilterDetailedResults
```
Comment (jhendrixMSFT) [inline sdk/ai/azopenai/models.go:1361]: This is a breaking change.

**Pull request merged**